test_api.py

```diff
@@ -880,126 +882,56 @@ def test_create_host_with_invalid_ansible_host(self):
 
                 self.verify_error_response(error_host, expected_title="Bad Request")
 
-    def test_create_host_with_20_byte_MAC_address(self):
-        system_profile = {
-            "network_interfaces": [{"mac_address": "00:11:22:33:44:55:66:77:88:99:aa:bb:cc:dd:ee:ff:00:11:22:33"}]
-        }
+    def test_create_host_with_tags_doesnt_work(self):
+        host_data = HostWrapper(test_data(tags=[{"namespace": "ns", "key": "some_key", "value": "val"}]))
+        create_response = self.post(HOST_URL, [host_data.data()], 207)
+        host_id = create_response["data"][0]["host"]["id"]
+        response = self.get(f"{HOST_URL}/{host_id}/tags")
 
-        host_data = HostWrapper(test_data(system_profile=system_profile))
+        self.maxDiff = None
+        self.assertEqual(response["results"][host_id], [])
 
-        response = self.post(HOST_URL, [host_data.data()], 207)
+    def test_update_host_with_tags_doesnt_change_tags(self):
+        host_data = HostWrapper(
+            test_data(tags=[{"namespace": "ns", "key": "some_key", "value": "val"}], fqdn="fqdn", id=generate_uuid())
+        )
 
-        self._verify_host_status(response, 0, 201)
+        message = {
+            "operation": "add_host",
+            "data": {
+                "id": host_data.id,
+                "account": host_data.account,
+                "display_name": host_data.display_name,
+                "tags": host_data.tags,
+                "reporter": host_data.reporter,
+                "stale_timestamp": host_data.stale_timestamp,
+                "facts": host_data.facts,
+                "fqdn": host_data.fqdn,
+            },
+        }
 
-        created_host = self._pluck_host_from_response(response, 0)
+        with self.app.app_context():
+            mock_event_producer = Mock()
+            handle_message(json.dumps(message), mock_event_producer)
+            json.loads(mock_event_producer.write_event.call_args[0][0])
 
-        original_id = created_host["id"]
+        # attempt to update
+        self.post(HOST_URL, [host_data.data()], 207)
```

Comment:
This does not try to change the tags, so we can’t distinguish the correct behavior (nothing changes) with the original one (tags are updated). You have to post a different tags and then check that even after the update the hosts still have the original set of tags. To achieve that it’d be also good to verify that the response is _200_ (updated) and not _201_ (created).

```suggestion
        update_host_data = HostWrapper(
            test_data(tags=[{"namespace": "other_ns", "key": "other_key", "value": "other_val"}], fqdn="fqdn")
        )
        update_response = self.post(HOST_URL, [update_host_data.data()], 207)
        self._verify_host_status(update_response, 0, 200)
```

It’s possible to get the host ID from here too.

```python
        updated_host = self._pluck_host_from_response(update_response, 0)
        host_id = updated_host["id"]
```